### PR TITLE
fix(mdtabs): Fix scrolling for tabs with a long last label

### LINKED ISSUE #4247: mdTabs: Cannot Scroll to the End of a Long Tab Label If It is the Last One
This is more apparent on mobile devices, but if you have a tab label that is longer than the viewport size, there is no way for the user to see the right edge of the last tab. This may be expected behaviour, but the right arrow will still show as "active" despite the fact that clicking the arrow will not actually do anything.

CodePen: http://codepen.io/anon/pen/rVEeeP

### PR DESCRIPTION
**Pull request by j-meyer**
Having the last tab longer than the canvas width will allow you to scroll to the end of the tab
rather than stop at the start of the tab.

Closes #4247

### CODE REVIEW

File: src/components/tabs/js/tabsController.js [hunk 1]
```diff
@@ -316,7 +316,11 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
       tab = elements.tabs[ i ];
       if (tab.offsetLeft + tab.offsetWidth > totalWidth) break;
     }
-    ctrl.offsetLeft = fixOffset(tab.offsetLeft);
+    if(i === elements.tabs.length - 1 && (tab.offsetLeft + tab.offsetWidth > totalWidth) && ctrl.offsetLeft === tab.offsetLeft){
```
Comment (EladBezalel) [inline src/components/tabs/js/tabsController.js]: Can you please explain this logic?

File: src/components/tabs/js/tabsController.js [hunk 2]
```diff
@@ -328,7 +332,11 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
       tab = elements.tabs[ i ];
       if (tab.offsetLeft + tab.offsetWidth >= ctrl.offsetLeft) break;
     }
-    ctrl.offsetLeft = fixOffset(tab.offsetLeft + tab.offsetWidth - elements.canvas.clientWidth);
+    if(tabs.offsetWidth > elements.canvas.clientWidth){
```
Comment (EladBezalel) [inline src/components/tabs/js/tabsController.js]: Here also

Comment (j-meyer) [inline src/components/tabs/js/tabsController.js]: Checks to see if the tab's length is greater than the available display space.

File: src/components/tabs/js/tabsController.js (hunk 1 above)
Comment (j-meyer) [inline src/components/tabs/js/tabsController.js]: It checks if you're on the left side of the last tab, and there isn't enough display space to show the entire tab, in which case you need to still scroll when the user hits the right arrow.

File: src/components/tabs/js/tabsController.js (hunk 2 above)
Comment (EladBezalel) [inline src/components/tabs/js/tabsController.js]: Please add it as comment

**Comment by EladBezalel** [comment]
Can you add tests?

**Comment by j-meyer** [comment]
I signed it!

**Comment by EladBezalel** [comment]
@j-meyer please add tests

**Comment by j-meyer** [comment]
I'll try to get to it this week, it's not the easiest thing to unit test as it requires checking the rendered Dom.

**Comment by EladBezalel** [comment]
@j-meyer feel free to reach me if you encounter any problem

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our deprecation effort.
    For details, see our post [Spring Cleaning 2016](https://groups.google.com/forum/#!topic/ngMaterial/3Ip9nd68bfk).

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Pull request closed without merging**